integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonDeleteStageCommand.scala

```diff
@@ -0,0 +1,192 @@
+    setAuditTable(table)
+    if (!table.getTableInfo.isTransactionalTable) {
+      throw new MalformedCarbonCommandException("Unsupported operation on non transactional table")
+    }
+    if (table.isChildTableForMV) {
+      throw new MalformedCarbonCommandException("Unsupported operation on MV table")
+    }
+    val tablePath = table.getTablePath
+    val startTime = System.currentTimeMillis()
+    val stageDataFileLocation = options.get("data_file_location")
+    if (stageDataFileLocation.isEmpty) {
+      throw new MalformedCarbonCommandException("Option [data_file_location] is not specified.")
+    }
+    val stageDataFileActiveTime = try {
+      Integer.valueOf(options.getOrElse("data_file_retain_time_second", "0")) * 1000
+    } catch {
+      case _: NumberFormatException =>
+        throw new MalformedCarbonCommandException(
+          "Option [data_file_retain_time_second] is not a number.")
+    }
+    if (stageDataFileActiveTime < 0) {
+      throw new MalformedCarbonCommandException(
+        "Option [data_file_retain_time_second] is negative.")
+    }
+    val stageDataFilesReferenced =
+      listStageDataFilesReferenced(listStageFiles(tablePath, configuration), configuration)
+    val stageDataFiles = listStageDataFiles(stageDataFileLocation.get, configuration)
+    stageDataFiles.map(
+      stageDataFile => {
+        // Which file will be deleted:
+        // 1. Not referenced by any stage file;
+        // 2. Has passed retain time.
+        if (stageDataFilesReferenced.contains(stageDataFile.getCanonicalPath)) {
+          null
+        } else if ((startTime - stageDataFile.getLastModifiedTime) < stageDataFileActiveTime) {
+          null
+        } else if (!stageDataFile.delete()) {
+          null
+        } else {
+          // Stage data file deleted.
+          stageDataFile.getCanonicalPath
+        }
+      }
+    ).filter(
+      stageDataFile => stageDataFile != null
+    ).map(
+      stageDataFile => Row.fromSeq(Seq(stageDataFile))
+    )
+    Seq.empty
+  }
+
+  private def listStageFiles(tablePath: String, configuration: Configuration): Seq[CarbonFile] = {
+    val stagePath = CarbonTablePath.getStageDir(tablePath)
+    val stageDirectory = FileFactory.getCarbonFile(stagePath, configuration)
+    if (stageDirectory.exists()) {
+      stageDirectory.listFiles().filter { file =>
+        !file.getName.endsWith(CarbonTablePath.SUCCESS_FILE_SUBFIX)
+      }
+    } else {
+      Seq.empty
+    }
+  }
+
+  private def listStageDataFiles(location: String,
+      configuration: Configuration): Seq[CarbonFile] = {
+    val stageDataFileLocation = FileFactory.getCarbonFile(location, configuration)
+    if (!stageDataFileLocation.exists()) {
+      LOGGER.warn("Stage data file location is not exists. " + location)
+      Seq.empty
+    } else {
+      stageDataFileLocation.listFiles(true).asScala
+    }
+  }
+
+  private def listStageDataFilesReferenced(stageFiles: Seq[CarbonFile],
```

Comment:
move `stageFiles: Seq[CarbonFile]` to next line

Reply:
Done.